required/amsmath/amsmath.dtx

```diff
@@ -1139,15 +1147,37 @@ Foreign command \@backslashchar#1;\MessageBreak
          \gdef\thedots@{\dotsb@}%
        \else
 %    \end{macrocode}
-% \changes{v2.15d}{2016/06/28}{Add space token to prevent runaway argument error}
-%    \begin{macrocode}
-         \xdef\meaning@{\meaning\@let@token. .........}%
-%    \end{macrocode}
-% In previous versions \verb|\long| macros were not seen by the lokkahead.
+%    In case \cs{\@let@token} is a robust \LaTeXe{} command, i.e.,
+%    expands to \cs{protect}
+%    \verb*=\somename =, we save the \cs{meaning} of
+%    \verb*=\somename = in \cs{meaning@} (possibly followed by some dots that have
+%    not been consumed as arguments during the expansion).
+%    Otherwise, we save the
+%    \cs{meaning} of \cs{@let@token} in \cs{meaning@}.  The dots at the end
+%    as well as the space are
+%    used in later parts of the checking (could probably be cleaned up a bit).
+% \changes{v2.15d}{2016/06/28}{Add space token to prevent runaway
+%    argument error}
+% \changes{v2.17r}{2024/06/29}{Drop \cs{protect} if present (gh/1265)}
+%    \begin{macrocode}
+       \begingroup
+         \def\protect{\protect}% % make it a quark
```

Comment:
Hope this is safe!  
Is it clearly local? -- Yes, good!

Reply:
:-)